camel/toolkits/ask_news_toolkit.py

```diff
@@ -0,0 +1,747 @@
+
+        Args:
+            query (str): The content of the user's message.
+            model (str): The model to use for generating the chat response.
+                (default: :obj:`"meta-llama/Meta-Llama-3-70B-Instruct"`)
+
+        Returns:
+            str: The content of the response message.
+
+        Raises:
+            Exception: If there is an error while processing the chat query.
+        """
+        try:
+            response = self.asknews_client.chat.get_chat_completions(
+                model=model,
+                messages=[{"role": "user", "content": query}],
+                stream=False,
+            )
+
+            # Return the content of the first choice's message
+            return response.choices[0].message.content
+
+        except Exception as e:
+            raise Exception(
+                f"An error occurred while processing the chat query: {e!s}."
+            )
+
+    def get_web_search(
+        self,
+        queries: List[str],
+        return_type: Literal["string", "dicts", "both"] = "string",
+    ) -> Union[str, dict, Tuple[str, dict]]:
+        r"""Perform a live web search based on the given queries.
+
+        Args:
+            queries (List[str]): A list of search queries.
+            return_type (Literal["string", "dicts", "both"]): The format of the
+                return value. (default: :obj:`"string"`)
+
+        Returns:
+            Union[str, dict, Tuple[str, dict]]: A string, dictionary, or both
+                containing the search results, or an error message if the
+                process fails.
+        """
+        try:
+            response = self.asknews_client.chat.live_web_search(
+                queries=queries
+            )
+
+            if return_type == "string":
+                search_content = response.as_string
+            elif return_type == "dicts":
+                search_content = response.as_dicts
+            elif return_type == "both":
+                search_content = (response.as_string, response.as_dicts)
+
+            return search_content
+
+        except Exception as e:
+            raise Exception(
+                "An error occurred while performing the web search for "
+                + f"'{queries}': {e!s}."
+            )
+
+    def get_tools(self) -> List[OpenAIFunction]:
+        r"""Returns a list of OpenAIFunction objects representing the functions
+          in the toolkit.
+
+        Returns:
+            List[OpenAIFunction]: A list of OpenAIFunction objects representing
+                the functions in the toolkit.
+        """
+        return [
+            OpenAIFunction(self.get_news),
+            OpenAIFunction(self.search_reddit),
+            OpenAIFunction(self.get_stories),
+            OpenAIFunction(self.finance_query),
+            OpenAIFunction(self.chat_query),
+            OpenAIFunction(self.get_web_search),
+        ]
+
+
+class AsyncAskNewsToolkit(BaseToolkit):
```

Comment:
Just curious, why do we need an async version of this toolkit

Reply:
This is supplemented based on @Wendong-Fan 's suggestion, merely to support possible future use, I guess.